libs/designer/src/lib/ui/templates/connections/workflowconnections.tsx

```diff
@@ -195,7 +200,7 @@ export const WorkflowConnections = ({ connections }: WorkflowConnectionsProps) =
       connection: connectionToUse ? { id: connectionToUse.id, displayName: getConnectionDisplayName(connectionToUse) } : undefined,
     });
 
-    if (!itemHasConnection && connectionToUse) {
+    if ((connectionToOverride || !itemHasConnection) && connectionToUse) {
```

Comment:
Do you need connectionToOverride here? because this is going to update everytime you change the tabs

Reply:
No... onConnectionsLoaded will be called whenever component is getting mounted that can be even when you are changing tabs

Reply:
I see. in that case, we need to have a flag to make sure we only override it for the first time. Is there a place where you would suggest to save that flag?

Reply:
Can you please what we do for basics and parameters override? We only save it onec in the store... here the component would always read the connection fore this item from store if it is not present it would look at override ... that is why i unresolved the previous comment because you need to change the order of that and remove connectionToOverride from here.

Reply:
Could you please elaborate what do you mean by save it only once? Currently we are saving it only once (that is when both manifest and connections are loaded, we calculate the override and save it). I've pinged you on IM and if needed let's sync up the details.

Reply:
resolved after syncing up offline.